### PR TITLE
Fix few paths of the products API

### PR DESCRIPTION
**Pull request by jisson**
Fix few paths of the products API client. Some were incomplete such as:
 - [POST] `crm-objects/v1/{product_id}` -> `crm-objects/v1/objects/products/{product_id}`

Cf: https://developers.hubspot.com/docs/methods/products/update-products

Also add `test_products`.

### CODE REVIEW

File: hubspot3/products.py [hunk 1]
```diff
@@ -72,17 +71,22 @@ def _get_path(self, subpath: str):
     def create(self, data=None, **options):
         """Create a new product."""
         data = data or {}
-
-        # See: https://developers.hubspot.com/docs/methods/products/create-product
-        data = [{"name": name, "value": value} for name, value in data.items()]
```
Comment (jisson) [inline hubspot3/products.py]: Note: I removed this as I felt that it was not a convention shared across the whole code base. I personally find more handy to be able to simply give a simple key / value `dict` but I wanted to be coherent with the rest of the code :)

File: hubspot3/products.py [hunk 2]
```diff
@@ -12,7 +12,6 @@
 class ProductsClient(BaseClient):
     """
     Products extension for products API endpoint
-    THIS API ENDPOINT IS ONLY A PREVIEW AND IS SUBJECT TO CHANGE
```
Comment (jisson) [inline hubspot3/products.py]: Is this still the case? I didn't see this warning on the product documentation

Comment (jpetrucciani) [reply hubspot3/products.py]: Looks like they've updated that - good catch!

**Comment by jpetrucciani** [comment]
Awesome - thanks for the contributions! I'll get these reviewed and merged soon!

File: hubspot3/products.py (hunk 1 above)
Comment (jpetrucciani) [reply hubspot3/products.py]: Hmm okay, that makes sense. I'll need to mention this change in the release notes

**Pull request merged**